File: python/LeetCode3068.py
class Solution:
    def maximumValueSum(self, nums: List[int], k: int, edges: List[List[int]]) -> int:
        count = 0
        min_delta = 10**9
        ans = 0

        for i in range(len(nums)):
            n = nums[i]
            m = nums[i] ^ k
            delta = m - n

            if m > n:
                count = (count + 1) % 2
                ans += m
                if delta < min_delta:
                    min_delta = delta
            else:
                ans += n
                if -delta < min_delta:
                    min_delta = -delta

        return ans if count == 0 else ans - min_delta

File: cpp/LeetCode3068.cpp
class Solution {
    public:
        long long maximumValueSum(vector<int>& nums, int k, vector<vector<int>>& edges) {
            long long ans = 0;
            int min_delta = 1000000000;
            int count = 0;
    
            for (int i = 0; i < nums.size(); i++) {
                int n = nums[i];
                int m = n ^ k;
                int delta = m-n;
    
                if (delta > 0) {
                    ans += m;
                    count = (count + 1) % 2;
                    if (delta < min_delta) {
                        min_delta = delta;
                    }
                } else {
                    ans += n;
                    if (-delta < min_delta) {
                        min_delta = -delta;
                    }
                }
            }
    
            if (count == 0) {
                return ans;
            } else {
                return ans - min_delta;
            }
            
        }
    };